v1/wd_cipher.c

```diff
@@ -467,10 +467,17 @@ int wcrypto_burst_cipher(void *ctx, struct wcrypto_cipher_op_data **opdata,
 int wcrypto_do_cipher(void *ctx, struct wcrypto_cipher_op_data *opdata,
 		void *tag)
 {
+	int ret;
+
 	if (!tag)
-		return wcrypto_burst_cipher(ctx, &opdata, NULL, 1);
+		ret = wcrypto_burst_cipher(ctx, &opdata, NULL, 1);
 	else
-		return wcrypto_burst_cipher(ctx, &opdata, &tag, 1);
+		ret = wcrypto_burst_cipher(ctx, &opdata, &tag, 1);
+
+	if (likely(ret == 1))
+		return WD_SUCCESS;
+
+	return ret;
 }
```

Comment:
Here is a complete modification method. I will send a patch tomorrow to fix this problem.